instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/Instrumenter.java

```diff
@@ -164,6 +165,10 @@ Context startAndEnd(
   }
 
   private Context doStart(Context parentContext, REQUEST request, @Nullable Instant startTime) {
+    return InstrumenterContext.withContext(() -> doStartImpl(parentContext, request, startTime));
```

Comment:
sorry for the delayed feedback, I wish this (relatively small) overhead didn't affect all instrumentations just for this edge case